addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/xml/InputConverterXML.java

```diff
@@ -0,0 +1,167 @@
+import java.util.*;
+import java.util.stream.Stream;
+
+import static java.util.stream.Collectors.toSet;
+import static org.openhab.binding.yamahareceiver.YamahaReceiverBindingConstants.*;
+
+/**
+ * XML implementation of {@link InputConverter}.
+ *
+ * @author Tomasz Maruszak - Input mapping fix
+ *
+ */
+public class InputConverterXML implements InputConverter {
+
+    private final Logger logger = LoggerFactory.getLogger(InputConverterXML.class);
+
+    private final WeakReference<AbstractConnection> comReference;
+
+    /**
+     * User defined mapping for state to input name.
+     */
+    private final Map<String, String> inputMap;
+    /**
+     * Holds all the inputs names that should NOT be transformed by the {@link #convertNameToID(String)} method.
+     */
+    private final Set<String> inputsWithoutMapping;
+
+    public InputConverterXML(AbstractConnection con, String inputMapConfig) {
+        this.comReference = new WeakReference<>(con);
+
+        logger.trace("User defined mapping: {}", inputMapConfig);
+        this.inputMap = createMapFromSetting(inputMapConfig);
+
+        try {
+            this.inputsWithoutMapping = createInputsWithoutMapping();
+            logger.trace("These inputs will not be mapped: {}", inputsWithoutMapping);
+        } catch (IOException | ReceivedMessageParseException e) {
+            throw new RuntimeException("Could not communicate with the device", e);
+        }
+    }
+
+    /**
+     * Creates a map from a string representation: "KEY1=VALUE1,KEY2=VALUE2"
+     *
+     * @param setting
+     * @return
+     */
+    private Map<String, String> createMapFromSetting(String setting) {
+        Map<String, String> map = new HashMap<>();
+
+        if (!StringUtils.isEmpty(setting)) {
+            String[] entries = setting.split(","); // will contain KEY=VALUE entires
+            for (String entry : entries) {
+                String[] keyValue = entry.split("="); // split the KEY=VALUE string
+                if (keyValue.length != 2) {
+                    logger.warn("Invalid setting: {} entry: {} - KEY=VALUE format was expected", setting, entry);
+                } else {
+                    String key = keyValue[0];
+                    String value = keyValue[1];
+
+                    if (map.putIfAbsent(key, value) != null) {
+                        logger.warn("Invalid setting: {} entry: {} - key: {} was already provided before", setting, entry, key);
+                    }
+                }
+            }
+        }
+        return map;
+    }
+
+    private Set<String> createInputsWithoutMapping() throws IOException, ReceivedMessageParseException {
+        // Tested on RX-S601D, RX-V479
+        Set<String> inputsWithoutMapping = Stream
+                .of(INPUT_SPOTIFY, INPUT_BLUETOOTH)
+                .collect(toSet());
+
+        Set<String> nativeInputNames = XMLProtocolService.getInputs(comReference.get(), Zone.Main_Zone)
+                .stream()
```

Comment:
I personally prefer this formatting but if you auto-format with the default eclipse it will break you could surround it with an hint for Eclipse to leave it alone :-)

Reply:
IntelliJ seems to preserve the formatting. Should I bother with no no formatting comment for eclipse?